File: lib/features/admin/data/models/admin_audit_log.dart
class AdminAuditLog {
  const AdminAuditLog({
    required this.id,
    required this.action,
    required this.adminName,
    required this.adminEmail,
    required this.targetType,
    required this.targetId,
    required this.description,
    required this.createdAt,
    this.ipAddress = '',
  });

  factory AdminAuditLog.fromJson(Map<String, dynamic> json) {
    final admin = json['adminUserId'];
    final adminMap = admin is Map ? Map<String, dynamic>.from(admin) : null;

    return AdminAuditLog(
      id: json['_id']?.toString() ?? '',
      action: json['action']?.toString() ?? '',
      adminName: adminMap?['name']?.toString() ?? '',
      adminEmail: adminMap?['email']?.toString() ?? '',
      targetType: json['targetType']?.toString() ?? '',
      targetId: json['targetId']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      ipAddress: json['ipAddress']?.toString() ?? '',
      createdAt: json['createdAt']?.toString() ?? '',
    );
  }

  final String id;
  final String action;
  final String adminName;
  final String adminEmail;
  final String targetType;
  final String targetId;
  final String description;
  final String ipAddress;
  final String createdAt;
}

File: backend/src/models/AdminAuditLog.js
import mongoose from 'mongoose';

const adminAuditLogSchema = new mongoose.Schema(
  {
    adminUserId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    action: { type: String, required: true, trim: true, index: true },
    targetType: { type: String, default: '', trim: true },
    targetId: { type: String, default: '', trim: true },
    description: { type: String, default: '', maxlength: 1000 },
    ipAddress: { type: String, default: '' },
    userAgent: { type: String, default: '' },
  },
  { timestamps: true }
);

adminAuditLogSchema.index({ targetType: 1, targetId: 1 });
adminAuditLogSchema.index({ createdAt: -1 });

const AdminAuditLog = mongoose.model('AdminAuditLog', adminAuditLogSchema);
export default AdminAuditLog;
